typing/printtyp.ml

```diff
@@ -1191,7 +1191,13 @@ let rec tree_of_type_decl id decl =
         decl.type_private
   in
   let immediate =
-    Builtin_attributes.immediate decl.type_attributes
+    match
+      Builtin_attributes.immediate decl.type_attributes,
+      Builtin_attributes.immediate64 decl.type_attributes
+    with
+    | true, _ -> Always
+    | false, true -> Always_on_64bits
+    | false, false -> Unknown
```

Comment:
Can this use `Typedecl_immediacy.marked_as_immediate` to avoid the duplication or does it create a cycle? Then, pulling `marked_as_immediate` into the suggested `Types_immediacy` can be a good idea. By the way, the `marked_as_immediate` name made more sense back when immediacy was boolean.

Reply:
Good idea. I did that